src/software/ai/navigator/path_manager/path_objective.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include <vector>
+
+#include "software/ai/navigator/obstacle/obstacle.h"
+
+/**
+ * Path objective is used to pass information from navigator
+ * to path manager to plan paths
+ */
+class PathObjective
+{
+   public:
+    PathObjective(const Point start, const Point end, const double current_velocity,
```

Comment:
Should the `current_velocity` be a vector? I'm not sure how useful the current speed of a robot is if we don't know which way it's moving.

Reply:
Currently we're using speed, but I agree velocity would be more useful. This PR is about re-architecture and refactoring, so we won't fix that in this PR.